warehouse/scripts/dbt_artifacts.py

```diff
@@ -160,17 +169,113 @@ def select(self) -> Select:
         ]
         return select(columns=columns)
 
+    @property
+    def num_bytes(self) -> Optional[int]:
+        if self.catalog_entry and "num_bytes" in self.catalog_entry.stats:
+            value = self.catalog_entry.stats["num_bytes"].value
+            # for some reason, 0 gets parsed as bool by pydantic
```

Comment:
There's no more graceful way we know of coercing this into the right type via something built into Pydantic?

Reply:
Ah yeah we could use a custom validator, I'll do that

Reply:
Ah custom validator doesn't make it easy to do the static typing though, so I just used a property on CatalogTable directly